guarddog/analyzer/metadata/pypi/repository_integrity_missmatch.py

```diff
@@ -0,0 +1,242 @@
+            return entry
+    # solution 1 did not work, let's be a bit more aggressive
+    for entry in clean_candidates:
+        owner, repo = extract_owner_and_repo(entry)
+        if repo is not None and (
+                # Idea: replace by if two strings have a Levenshtein distance < X% of string length
+                repo.lower() in name.lower() or name.lower() in repo.lower()):
+            return entry
+    return None
+
+
+def dict_generator(indict, pre=None):
+    pre = pre[:] if pre else []
+    if isinstance(indict, dict):
+        for key, value in indict.items():
+            if isinstance(value, dict):
+                for d in dict_generator(value, pre + [key]):
+                    yield d
+            elif isinstance(value, list) or isinstance(value, tuple):
+                for v in value:
+                    for d in dict_generator(v, pre + [key]):
+                        yield d
+            else:
+                yield pre + [key, value]
+    else:
+        yield pre + [indict]
+
+
+def get_file_hash(path):
+    with open(path, 'rb') as f:
+        # Read the contents of the file
+        file_contents = f.read()
+        # Create a hash object
+        hash_object = hashlib.sha256()
+        # Feed the file contents to the hash object
+        hash_object.update(file_contents)
+        # Get the hexadecimal hash value
+        return hash_object.hexdigest(), str(file_contents).strip().splitlines()
+
+
+def _ensure_proper_url(url):
+    parsed = urllib3.util.parse_url(url)
+    if parsed.scheme is None:
+        url = f"https://{url}"
+    return url
+
+
+def find_github_candidates(package_info) -> Tuple[set[str], Optional[str]]:
+    infos = package_info["info"]
+    homepage = None
+    if "Homepage" in package_info["info"]["project_urls"]:
+        homepage = package_info["info"]["project_urls"]["Homepage"]
+    github_urls = set()
+    for dict_path in dict_generator(infos):
+        leaf = dict_path[-1]
+        if type(leaf) is not str:
+            continue
+        res = re.findall(GH_REPO_REGEX, leaf)
+        if len(res) > 0:
+            for cd in res:
+                github_urls.add(_ensure_proper_url(cd.strip()))
+    best = None
+    if homepage in github_urls:
+        best = _ensure_proper_url(homepage)
+    return github_urls, best
+
+
+def exclude_result(file_name, repo_root, pkg_root):
+    if file_name.endswith('.rst') or file_name.endswith('.md'):
+        return True
+    if file_name.endswith("setup.cfg"):
+        repo_cfg = configparser.ConfigParser()
+        repo_cfg.read(os.path.join(repo_root, file_name))
+        pkg_cfg = configparser.ConfigParser()
+        pkg_cfg.read(os.path.join(pkg_root, file_name))
+        repo_sections = list(repo_cfg.keys())
+        pkg_sections = list(pkg_cfg.keys())
+        if "egg_info" in pkg_sections and "egg_info" not in repo_sections:
+            return True
+    return False
+
+
+def find_missmatch_for_tag(repo, tag, base_path, repo_path):
```

Comment:
As we chatted earlier we might not want to have this tag-based logic already? It's only adapted to a very specific release process that not everyone follows (?)

Reply:
I think we can still use it for the v0 and then iterate on a bisect algorithm that looks up for the best commit, wdyt?

Reply:
Yes, let's keep as is for now